### PR TITLE
fmt: Do not try and format missing Haskell files.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
And format fewer Haskell and Scala files when not fully rebased from trunk.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
Spotted this when trying to open a PR against the 1.6 release branch. This was trying to pass a file that didn't exist on that branch to `hlint`.

File: fmt.sh [hunk 1]
```diff
@@ -110,12 +110,13 @@ run dade-copyright-headers "$dade_copyright_arg" .
 # We do test hlint via Bazel rules but we run it separately
 # to get linting failures early.
 if [ "$hlint_diff" = "true" ]; then
-    changed_haskell_files="$(git diff --name-only origin/master | grep '\.hs$' || [[ $? == 1 ]])"
-    if [ "" != "$changed_haskell_files" ]; then
-        hlint -j4 $changed_haskell_files
-    fi
+  #  We do not run on deleted files, or files that have been added since we last rebased onto trunk.
+  changed_haskell_files="$(git diff --name-only --diff-filter=d "$(git merge-base origin/master HEAD)" | grep '\.hs$' || [[ $? == 1 ]])"
```
Comment (garyverhaegen-da) [inline fmt.sh:114]: Any opinion on `--diff-filter=AM`?

**Review by garyverhaegen-da — APPROVED** [review]

File: fmt.sh (hunk 1 above)
Comment (SamirTalwar-DA) [reply fmt.sh:114]: I'm fine with that. It has the added benefit of not running on untracked files, which shouldn't be a problem.

Comment (SamirTalwar-DA) [reply fmt.sh:114]: We probably want `--diff-filter=ACMRT`.

Comment (garyverhaegen-da) [reply fmt.sh]: :+1:

**Pull request merged**